src/main/java/strategy/PurchaseOperationHandler.java

```diff
@@ -0,0 +1,21 @@
+package strategy;
+
+import java.util.Map;
+
+public class PurchaseOperationHandler implements OperationHandler {
+    @Override
+    public void apply(Map<String, Integer> inventory, String fruit, int quantity) {
+        int currentQuantity = inventory.getOrDefault(fruit, 0);
+        if (currentQuantity < quantity) {
+            throw new RuntimeException("Not enough " + fruit + " in inventory.");
+        }
```

Comment:
This check for not having enough fruit in the inventory is correct, but throwing a RuntimeException with a generic message is not informative enough. Consider throwing a custom exception that extends RuntimeException for a more descriptive error handling. For example, create an `OperationException` with a message that indicates a purchase operation failed due to insufficient stock.